runtime-modules/project-token/src/migrations.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+use frame_support::traits::OnRuntimeUpgrade;
+
+// syntactic sugar for logging.
+#[macro_export]
+macro_rules! log {
+	($level:tt, $patter:expr $(, $values:expr)* $(,)?) => {
+		log::$level!(
+			target: $crate::LOG_TARGET,
+			concat!("[{:?}] 📹 ", $patter), <frame_system::Pallet<T>>::block_number() $(, $values)*
+		)
+	};
+}
+
+pub mod nara {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+
+    pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+    impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+            let onchain = Pallet::<T>::on_chain_storage_version();
+
+            ensure!(onchain < 1, "this migration can be deleted");
+
+            let module_account_id = Module::<T>::module_treasury_account();
+            let deposit = T::JoyExistentialDeposit::get();
+            let treasury_free_balance = Joy::<T>::free_balance(&module_account_id);
+
+            ensure!(treasury_free_balance >= Joy::<T>::existential_deposit());
```

Comment:
Does it make sense to assert this? The whole point is that this is a new account which would have no free balance.. So this will fail when run against mainnet chain..If anything you would be asserting the free balance is in-fact zero.

Reply:
Besides that, the ensure!() macro needs a second argument which is the error if assertion fails.